packages/frontend-2/plugins/survicate.ts

```diff
@@ -0,0 +1,86 @@
+import {
+  useOnAuthStateChange,
+  useGetInitialAuthState
+} from '~/lib/auth/composables/auth'
+import { useActiveUser } from '~~/lib/auth/composables/activeUser'
+import { useSynchronizedCookie } from '~/lib/common/composables/reactiveCookie'
+import dayjs from 'dayjs'
+
+export default defineNuxtPlugin(async (app) => {
+  const { isLoggedIn } = useActiveUser()
+
+  const onboardingOrFeedbackDateString = useSynchronizedCookie<string | undefined>(
+    'onboardingOrFeedbackDate',
+    {
+      default: () => new Date().toISOString().split('T')[0],
+      expires: dayjs().add(999, 'day').toDate()
+    }
+  )
+
+  const onboardingOrFeedbackDate = onboardingOrFeedbackDateString.value
+    ? new Date(onboardingOrFeedbackDateString.value)
+    : new Date()
+
+  const shouldShowSurvey = checkSurveyDisplayConditions(onboardingOrFeedbackDate)
+
+  if (!process.client || !isLoggedIn.value || !shouldShowSurvey) {
```

Comment:
if you don't need this to run on the serverside, add a .client.ts prefix to the plugin